### PR TITLE
Isabel Suchanek -- Carets

### PR DESCRIPTION
**Pull request by isabeldepapel**
# Media Ranker
Congratulations! You're submitting your assignment!

## Comprehension Questions
Question | Answer
---      | ---
Describe a custom model method you wrote. | I wrote a self method for my Work model that generates a hash of the media with the type as the key, and an array of the works as the values with the works sorted in descending order by vote count. It iterates through the Category table (I separated it out bc I wanted practice with relational tables)--so if more categories got added, it would still work (but the view would break; a hash is unsorted so I couldn't figure out a way to fix the view display w/o hard coding)--sets the category name as the key, and then calls an ActiveRecord query using that category name (this is a helper method) and sets those results as the value. This keeps the controller skinny.
Describe how you approached testing that model method. What edge cases did you come up with? | This method depends on another helper method, so I tested the helper method first to make sure I would get a hash, and it would correctly select the works by category. So for this method I needed to test that it would return the works sorted by vote count, so I added votes to make sure the order of the works was changing accordingly. And then also verifying that it still worked if there were no works and no votes. 
Describe an edge case test you wrote for a controller | For logging in the user on the Users controller, I checked that if the login name was invalid (e.g. empty string), that it would return a bad request, and also that no session id had been created.
What are `session` and `flash`? What is the difference between them? | They're hashes that keep track of data. Session persists until the user closes the browser (or logs out), and flash persists through the next request (or flash now, which only persists on that request)
Describe a controller filter you wrote. | Find work, which finds the work by its id, and then I use it before edit, update, show, and destroy.
What was one thing that you gained more clarity on through this assignment? | Nested routes
What is the Heroku URL of your deployed application | https://salty-ocean-56796.herokuapp.com/
Do you have any recommendations on how we could improve this project for the next cohort? | A list of the flash messages and where the pages are supposed to redirect to would be nice

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+end
+
+group :development do
+  # Access an IRB console on exception pages or by using <%= console %> anywhere in the code.
+  gem 'web-console', '>= 3.3.0'
+  gem 'listen', '>= 3.0.5', '< 3.2'
+  # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
+  gem 'spring'
+  gem 'spring-watcher-listen', '~> 2.0.0'
+  gem 'guard-minitest', '~> 2.3.2'
```
Comment (CheezItMan) [inline Gemfile:51]: Neat use of guard!

**Comment by CheezItMan** [comment]
# Media Ranker
## What We're Looking For

Feature | Feedback
---     | ---
**Core Requirements** |
Git hygiene |  Good number of commits and good commit messages |
Comprehension questions	| Check, the flash messages are intended to push you to explore the app and figure out where you should notify the user of things |
**General** |
Rails fundamentals (RESTful routing, use of named paths) | Check, good restful routes |
Semantic HTML | Good semantic HTML |
Errors are reported to the user |  Check |
Business logic lives in the models | Nice work using SQL and SQL(ish) commands to get the lists sorted and arranged.  |
Models are thoroughly tested, including relations, validations and any custom logic |  Very through model testing |
Controllers are thoroughly tested |  Some really outstanding testing here, very nice work |
**Wave 1 - Media** |
Splash page shows the three media categories | Check
Basic CRUD operations on media are present and functional | Check
**Wave 2 - Users and Votes** |
Users can log in and log out | Check
The ID of the current user is stored in the session | Check
Individual user pages and the user list are present | Check
A user cannot vote for the same media more than once | Check
All media lists are ordered by vote count | Check
Splash page contains a media spotlight | Check
Media pages contain lists of voting users | Check
**Wave 3 - Styling** |
Foundation is used appropriately | Check, pretty good responsiveness.  The menu works fairly well as the screen shinks.
Look and feel is similar to the original | Check, looks very much like the original site
**Overall** |  This was a tough week and you turned in a good solid project.  You git all the requirements.  |
